forte/pipeline.py

```diff
@@ -222,6 +222,64 @@ def __init__(
         # Indicate whether do type checking during pipeline initialization
         self._do_init_type_check: bool = do_init_type_check
 
+    def find_entry_inherit(self, entry, filter_list: List[str]):
+        r"""Find all sub_classes of Entry Class.
+
+        Args:
+            entry: Entry class.
+            filter_list: Make sure the parent entry does not start with one of them.
+        """
+        spec_dict: Dict = {"definitions": []}
+        all_attributes_dict = {}
+        for subclass in entry.__subclasses__():
+            all_subclasses, all_attributes = self.find_spec_dict(subclass)
+            spec_dict["definitions"].extend(all_subclasses)
+            all_attributes_dict.update(all_attributes)
+
+        if len(filter_list) > 0:
+            for filter_name in filter_list:
+                spec_dict["definitions"] = [
+                    i
+                    for i in spec_dict["definitions"]
+                    if not i["parent_entry"].startswith(filter_name)
+                ]
+
+        return spec_dict, all_attributes_dict
+
+    def find_spec_dict(self, entry_subclass):
+        r"""
+        Recursive Method to Find all the Subclasses of a Class Given its Name.
+
+        Args:
+            entry_subclass: Entry Class.
+        """
+        all_subclasses = []
+        all_attributes: Dict = {}
+        if len(entry_subclass.__subclasses__()) > 0:
+            all_attributes[entry_subclass.__name__] = {}
+        parent_entry = entry_subclass.__module__ + "." + entry_subclass.__name__
+        for subclass in entry_subclass.__subclasses__():
+            entry_name = subclass.__module__ + "." + subclass.__name__
+            if (
+                not len(entry_name.split(".")) < 3
+                and not len(parent_entry.split(".")) < 3
+            ):
+                if "__annotations__" in subclass.__dict__.keys():
```

Comment:
Use try..except to save time

Reply:
If you have to import all realted types, import them in pipeline.py instead of the test file. You may not want users to import all classes every time they use this function. I tested locally and this should work.------------ At first, for user convenience, I imported them to pipeline.py. Yes, it works locally, but it could not pass the PyLint check in the PR, as shown in the figure below, so I imported them in the test case.
<img width="898" alt="image" src="https://user-images.githubusercontent.com/11881183/154897361-2ec14062-ec8f-4adc-9811-44ad13156252.png">